src/Context/ApiContext.php

```diff
@@ -273,15 +273,21 @@ private function dropSessionContext()
      */
     public function ensureSessionActive()
     {
-        if (is_null($this->sessionContext)) {
-            return;
+        if (!is_null($this->sessionContext) && $this->isExpired()) {
+            $this->resetSession();
         }
+    }
 
+    /**
+     * Checks if the session has expired
```

Comment:
Perhaps rename this function to something like `isSessionExpired` which makes this comment unnecessary.

Reply:
🤔 Fair enough will make this change in all the SDK's then 👍